include/eve/module/core/regular/bit_ternary.hpp

```diff
@@ -0,0 +1,389 @@
+#pragma once
+
+#include <eve/arch.hpp>
+#include <eve/traits/overload.hpp>
+#include <eve/module/core/decorator/core.hpp>
+#include <eve/module/core/regular/bit_and.hpp>
+#include <eve/module/core/regular/bit_andnot.hpp>
+#include <eve/module/core/regular/bit_or.hpp>
+#include <eve/module/core/regular/bit_notor.hpp>
+#include <eve/module/core/regular/bit_select.hpp>
+#include <eve/module/core/regular/bit_xor.hpp>
+#include <iostream>
+
+namespace eve
+{
+  template<typename Options>
+  struct bit_ternary_t : callable<bit_ternary_t, Options>
+  {
+    template<int K, value T0, value T1, value T2>
+    requires(eve::same_lanes_or_scalar<T0, T1, T2> && 0 <= K && K <= 255)
+      EVE_FORCEINLINE constexpr bit_value_t<T0, T1, T2>
+    operator()( std::integral_constant<int, K> k, T0 t0, T1 t1, T2 t2) const noexcept
+    {
+      return EVE_DISPATCH_CALL(k, t0,  t1, t2);
+    }
+
+    EVE_CALLABLE_OBJECT(bit_ternary_t, bit_ternary_);
+  };
+
+//================================================================================================
+//! @addtogroup core_bitops
+//! @{
+//!   @var bit_ternary
+//!   @brief `strict_elementwise_callable` object implementing ternary logic.
+//!
+//!   @groupheader{Header file}
+//!
+//!   @code
+//!   #include <eve/module/core.hpp>
+//!   @endcode
+//!
+//!   @groupheader{Callable Signatures}
+//!
+//!   @code
+//!   namespace eve
+//!   {
+//!      // Regular overloads
+//!      constexpr auto bit_ternary(integral_constant k, auto a, auto b,  auto c)         noexcept; // 1
+//!      constexpr auto bit_ternary(bit_tern k, auto a, auto b,  auto c)                  noexcept; // 2
+//!
+//!      // Lanes masking
+//!      constexpr auto bit_xor[conditional_expr auto c](/*any of the above overloads*/)  noexcept; // 3
+//!      constexpr auto bit_xor[logical_value auto m](/*any of the above overloads*/)     noexcept; // 3
+//!   }
+//!   @endcode
+//!
+//!   **Parameters**
+//!
+//!     * `k`: integral constant whose bits are the resultat of a lookup table for the ternary operation.
+//!       The consteval function mfb (make from bits) can be use to generate this constant from its bits (see below).
+//!     * `x`, `y`, `z`, : first [argument](@ref eve::value).
+//!     * `c`: [Conditional expression](@ref conditional_expr) masking the operation.
+//!     * `m`: [Logical value](@ref logical) masking the operation.
+//!
+//!    **Return value**
+//!
+//!      1. Bitwise ternary logic that provides the capability to implement any three-operand binary function;
+//!        the specific binary function is specified by value of k.
+//!        For each bit in each integer, the corresponding bit from a, b, and c are used according
+//!        to ik, and the result is output.
+//!      2. equivalent to the call on the elements of the tuple.
+//!      3. [The operation is performed conditionnaly](@ref conditional).
+//!
+//!      This is the pattern of a truth table:
```

Comment:
This piece of the doc is a bit unclear and need work

Reply:
done